ops/pebble.py

```diff
@@ -1969,6 +1986,88 @@ def read1(self, n: int = -1) -> Union[str, bytes]:
         return self.read(n)
 
 
+class IdentityAccess(enum.Enum):
+    """Enum of identity access levels."""
+
+    ADMIN = 'admin'
+    READ = 'read'
+    METRICS = 'metrics'
+    UNTRUSTED = 'untrusted'
+
+
+@dataclasses.dataclass
+class LocalIdentity:
+    """Local identity configuration (for ucrednet/UID authentication)."""
+
+    user_id: int
+
+    @classmethod
+    def from_dict(cls, d: LocalIdentityDict) -> LocalIdentity:
+        """Create new LocalIdentity from dict parsed from JSON."""
+        return cls(user_id=d['user-id'])
+
+
+@dataclasses.dataclass
+class BasicIdentity:
+    """Basic identity configuration (for HTTP basic authentication)."""
+
+    password: str  # sha512-crypt-hashed password
+
+    @classmethod
+    def from_dict(cls, d: BasicIdentityDict) -> BasicIdentity:
+        """Create new BasicIdentity from dict parsed from JSON."""
+        return cls(password=d['password'])
+
+
+@dataclasses.dataclass
+class Identity:
+    """Pebble identity configuration."""
+
+    access: Union[IdentityAccess, str]
```

Comment:
I think for new types we should just make this `access: IdentityAccess`. We've regretted using the union for existing types.

Reply:
The idea with the `str, Enum`-style enum was that we can now simplify this type to `access: IdentityAccess` as `IdentityAccess` is a subclass of `str`.